Convert-ProwlarrSupportedIndexersToMarkdownTable.ps1

```diff
@@ -1,108 +1,194 @@
-$fmttableuse= "## Usenet" + "`r`n`r`n" + "### Public" + "`r`n`r`n" + $fmttablePubUse + "`r`n`r`n" + "### Private" + "`r`n`r`n" + $fmttablePrvUse
-$date = [DateTime]::UtcNow.ToString("o")
-$mdHeader = "---
+### Private Torrents
+Write-Host 'Building: Torrents - Private'
+$tbl_PrvTor = $indexer_tbl_obj | Where-Object { ($_.privacy -CIn 'private' -and $_.protocol -eq 'torrent') } | `
+    Select-Object @{Name = 'Indexer'; Expression = $indexer_name_exp }, `
+@{Name = 'Language'; Expression = { $_.language } }, `
+@{Name = 'Description'; Expression = { $_.description } }         
+
+## Convert Data to Markdown Table
+#$tbl_fmt_PubUse= ConvertTo-Markdown($tbl_PubUse)
+$tbl_fmt_PubUse = 'None'
+$tbl_fmt_PrvUse = $tbl_PrvUse | Format-MarkdownTableTableStyle Indexer, Description, Language -HideStandardOutput -ShowMarkdown -DoNotCopyToClipboard
+$tbl_fmt_PubTor = $tbl_PubTor | Format-MarkdownTableTableStyle Indexer, Description, Language -HideStandardOutput -ShowMarkdown -DoNotCopyToClipboard 
+$tbl_fmt_PrvTor = $tbl_PrvTor | Format-MarkdownTableTableStyle Indexer, Description, Language -HideStandardOutput -ShowMarkdown -DoNotCopyToClipboard
+Write-Host 'Builds Converted to Markdown Tables'
+
+## Page Header Info
+$wiki_page_start = $wiki_1newline + "- Supported Indexers as of Build ``" + $build + "`` / [Commit: " + $commit + '](' + $wiki_commiturl + $commit + ')'
+Write-Host 'Page Header Built'
+
+## Build Page Pieces'
+$tbl_fmt_tor = $wiki_1newline + '## Torrents' + $wiki_2newline + '### Public' + $wiki_2newline + $tbl_fmt_PubTor + $wiki_2newline + '### Private' + $wiki_2newline + $tbl_fmt_PrvTor + $wiki_2newline
+$tbl_fmt_use = '## Usenet' + $wiki_2newline + '### Public' + $wiki_2newline + $tbl_fmt_PubUse + $wiki_2newline + '### Private' + $wiki_2newline + $tbl_fmt_PrvUse
+Write-Host 'Wiki Markdown Tables Built'
+$date = [DateTime]::UtcNow.ToString('o')
+$mdHeader = 
+"---
 title: Prowlarr Supported Indexers
-description: Indexers currently named as supported in the current nightly build of Prowlarr. Other indexers are available via either Generic Newznab or Generic Torznab. 
+description: Indexers currently named as supported in the current nightly build of Prowlarr. Other indexers are available via either Generic Newznab or Generic Torznab.
 published: true
 date: $date
 tags: prowlarr, indexers
 editor: markdown
 dateCreated: $date
 ---"
-
-$fmttableall= $mdHeader + "`n" + $prefix + "`n" + $fmttabletor + $fmttableuse
-$fmttableall | Out-File $OutputFile
+Write-Host 'Wiki Page pieces built'
+## Build and output Page
+$wiki_page_file = $mdHeader + $wiki_2newline + $wiki_page_start + $wiki_1newline + $tbl_fmt_tor + $tbl_fmt_use
+Write-Host 'Wiki Page Built'
+#$wiki_page_file | Out-File $OutputFile -Encoding $wiki_encoding
+[IO.File]::WriteAllLines(($OutputFile | Resolve-Path), $wiki_page_file)
```

Comment:
Why use the dotnet method and not just Out-File as per Line 192?

Reply:
this avoids the BOM being wrote

Reply:
This is only relevant on PowerShell 5.1 and below - Modern versions of PowerShell (6 and onwards) output UTF8 without BOM by default. PowerShell 7 is what is used within the build process. You should really install PowerShell 7 for development of this.

Reply:
fixed up